### PR TITLE
Handling boolean flags in ModelArguments

### LINKED ISSUE #175: [Feat]:  `ModelArguments` support boolean CLI flags
### Feature Type

- [x] Adding new functionality

- [x] Changing existing functionality

- [ ] Removing existing functionality

### Problem Description

When `ModelArguments` builds the CLI arguments for a config class with boolean attributes it expects them to be passed as parameters instead of a flag argument.

### Feature Description

Update `ModelArguments.add_arguments` to have the action of the argument to be "store_true", or "store_false", for a boolean value dependent on the default.

This is to allow the argument to be treated as a flag e.g. "--arg" instead of "--arg true".

### Alternative Solutions

Currently works by user passing "--arg true" as the argument.

### Additional Context

[ArgumentParser.add_argument docs](https://docs.python.org/3/library/argparse.html#the-add-argument-method)

### PR DESCRIPTION
**Pull request by MattBuckley-TfN**
# Changes

Updated `ModelArguments`  to use "store_true", and "store_false", to have flags for boolean values instead of requiring true / false  arguments.

- Closes #175 

# Tasks

- [x] Have unittests been added (testing individual functions and their edge cases)
- [x] Has documentation (including user guide) been updated
- [x] Have new dependencies been added (or changed) in relevant `requirements.txt`
- [x] Code linting, tests and other workflows pass

----
📚 Documentation preview 📚: https://caftoolkit--176.org.readthedocs.build/en/176/

### CODE REVIEW

File: src/caf/toolkit/arguments.py [hunk 1]
```diff
@@ -174,7 +175,7 @@ def parse_arg_details(annotation: str) -> tuple[type, bool, int | str | None]:
     """
     annotation = _replace_union(annotation)
 
-    match = re.match(r"^(?:(\w+)?\[)?([\w \t,.|]+)\]?$", annotation.strip())
+    match = re.match(r"^(?:(\w+)?\[|<class ')?([\w \t,.|]+)(?:\]|'>)?$", annotation.strip())
```
Comment (BenTaylor-TfN) [inline src/caf/toolkit/arguments.py:178]: Do the docs on the accepted types need updating in this function too?

**Review by BenTaylor-TfN — COMMENTED** [review]
How are you testing this works? Is it worth adding a quick test to ensure this works correctly?

File: src/caf/toolkit/arguments.py (hunk 1 above)
Comment (MattBuckley-TfN) [reply src/caf/toolkit/arguments.py:178]: Still only accepts str but I've updated the docstring to state it can handle `"<class 'bool'>"`

**Comment by MattBuckley-TfN** [comment]
> How are you testing this works? Is it worth adding a quick test to ensure this works correctly?

Added some extra correct tests for the type repr ("<class 'bool'>") for bool, int, float and str

**Review by BenTaylor-TfN — APPROVED** [review]
